Main/Source/Effort.Shared/Provider/EffortRestorePoint.cs

```diff
@@ -28,21 +30,36 @@ public void AddToIndex(object table, List<object> entities)
             }
         }
 
-        public void Restore(DbContext context)
+        public void Restore(DbContext context, object dbContainer)
         {
             if (OrderedEntities == null)
             {
                 CreateOrderedEntities();
                 EffortConnection.ClearTables(context);
             }
 
+            Dictionary<IExtendedTable, bool> oldIdentityFieldDictionary = new Dictionary<IExtendedTable, bool>();
+            if (dbContainer != null)
+            {
+                foreach (IExtendedTable table in ((DbContainer)dbContainer).Internal.Tables.GetAllTables())
```

Comment:
This code should be called before a call of CreateOrderedEntities, because it is first place where "insert" to table is called. Otherwise a next identity value will be used for entity. In bug #194  it can be seeing in error stack trace:
```
System.Exception: System.Exception: Oops! There is an error when trying to generate the insert order. ---> System.Reflection.TargetInvocationException: Адресат вызова создал исключение.(The target of an invocation has generated an exception.) ---> NMemory.Exceptions.ForeignKeyViolationException: Foreign key violation [dbo_DocumentType :: ProjectId]. The key value [1] does not exists in the referenced table [dbo_Projects :: Id].. Error code: RelationError.

Stack trace:
   in Effort.Shared.Provider.EffortRestorePoint.CreateOrderedEntities() <======
   in Effort.Shared.Provider.EffortRestorePoint.Restore(DbContext context)
   in Effort.Provider.EffortConnection.RollbackToRestorePoint(DbContext context)
```